File: pe-lwjgl-binding/src/main/java/org/saar/lwjgl/opengl/shaders/uniforms/Vec2iUniformValue.java
package org.saar.lwjgl.opengl.shaders.uniforms;

import org.joml.Vector2i;
import org.joml.Vector2ic;

public class Vec2iUniformValue extends Vec2iUniform implements UniformValue<Vector2ic> {

    private final String name;

    private final Vector2i value = new Vector2i();

    public Vec2iUniformValue(String name) {
        this.name = name;
    }

    @Override
    public final Vector2ic getUniformValue() {
        return getValue();
    }

    @Override
    public final String getName() {
        return this.name;
    }

    @Override
    public final Vector2i getValue() {
        return this.value;
    }

    @Override
    public final void setValue(Vector2ic value) {
        getValue().set(value);
    }
}

File: pe-lwjgl-binding/src/main/java/org/saar/lwjgl/opengl/shaders/uniforms/Vec2iUniformValue.kt
package org.saar.lwjgl.opengl.shaders.uniforms

import org.joml.Vector2i
import org.joml.Vector2ic
import org.saar.maths.JomlDelegates

class Vec2iUniformValue(
    override val name: String,
    initialValue: Vector2ic = Vector2i(0, 0)
) : Vec2iUniform() {
    override var value: Vector2ic by JomlDelegates.CachedVector2i(Vector2i(initialValue))
}